src/ngraph/op/broadcast.cpp

```diff
@@ -144,6 +145,31 @@ void op::v1::Broadcast::validate_and_infer_types()
         result_shape = static_pointer_cast<op::Constant>(input_value(1).get_node_shared_ptr())
                            ->get_shape_val();
     }
+    else if (dynamic_pointer_cast<op::Concat>(input_value(1).get_node_shared_ptr()))
```

Comment:
Specify the op version by op version or opset. `op::Concat` could change over time.
Use `else if (auto concat = as_type_ptr<opset1::Concat>(...)`

Reply:
Done

Reply:
As you suggested in Slack, let's use `op::v0::Concat` here.